modules/dhcp/dhcp_api.rb

```diff
@@ -16,17 +16,28 @@ class Proxy::DhcpApi < ::Sinatra::Base
         end
         @server = Proxy::DHCP::ISC.new(:name => "127.0.0.1",
                                        :config => Proxy::DhcpPlugin.settings.dhcp_config,
-                                       :leases => Proxy::DhcpPlugin.settings.dhcp_leases)
+                                       :leases => Proxy::DhcpPlugin.settings.dhcp_leases,
+                                       :service => Proxy::DHCP::SubnetService.new(Proxy::MemoryStore.new, Proxy::MemoryStore.new,
```

Comment:
Could these be optimised so it's not repeated for each provider?  Perhaps by initialising sane defaults inside the Proxy::DHCP::SubnetService.